jvm/src/Language/Java.hs

```diff
@@ -416,6 +427,9 @@ class (Interp (Uncurry a) ~ ty, SingI ty, IsReferenceType ty)
       => Reflect a ty where
   reflect :: a -> IO (J ty)
 
+  default reflect :: Coercible a ty => a -> IO (J ty)
+  reflect x = newLocalRef (jobject x)
+
```

Comment:
I was in the car when I wanted to draw your attention to this earlier - I'm considering removing those two instances, now that we have the defaults. That's why I didn't use those instances in the definition.

It always bugged me that a local `J ty` and a global `J ty` have the same type. With the default definitions we have now, we could say that anything that's wrapped is owned Haskell side, and anything not is normally owned Java side only. Thoughts?